GVFS/GVFS.Common/Tracing/TelemetryDaemonEventListener.cs

```diff
@@ -112,6 +112,13 @@ private void SendMessage(string message)
 
             try
             {
+                // If we're in byte/stream transmission mode rather than message mode
+                // we should signal the end of each message with a line-feed (LF) character.
+                if (this.pipeClient.TransmissionMode == PipeTransmissionMode.Byte)
+                {
+                    message += '\n';
```

Comment:
If we're using a single '\n' for end of message, how much risk is there a message would contain a linefeed in the message payload?

Reply:
The json formatter should backslash encode LF chars that are inside string fields.
The json formatter will only insert LFs in "indented/pretty" formatting mode.

So I think we're good here.  But I can't see the caller in this view, so we should test this.
I have stuff in the GItService to discard json fragments, so if I get one, we'll ignore the message.

Reply:
I did wonder about this too.. perhaps we should just play it safer by using something like two null-bytes? \0\0?

Reply:
let's wait on this for now.  

let's do some testing where we build a dictionary with embedded LFs and use the
existing ETW call stack to send the dictionary and see what we get.

If we do manage to get a raw (un-escaped) LF onto the wire, then we can look at
making it a single NULL byte.  (A 2 byte sequence changes the message decoder
state machine (not a big deal, but just more work (that i'm not sure we need yet)).)

Reply:
Just did a quick test; I added a bunch of LFs to the event name, and to the 'payload' dictionary (both key and value). The JSON payload is indeed escaped correctly and looks like this in the debug console for the GitService:

```
...\"\\n\\nLINE-FEED\\n\\n\":\"\\n\\n\\n\\n\\n\\n\"}"}}
```

The event name however seemed to be printed literally, with the line-feeds resulting in a new line as shown in the debug console:

```
Microsoft.Git.GVFS/EnlistmentInfo

: {"version":{"service":"1.0.1.0","vfs":"0.2.173.2"},...
```

whereas normal messages look like:

```
Microsoft.Git.GVFS/PreMount: {"version":{"service":"1.0.1.0"...
```

..but I think this is just aesthetic; the messages are being received and decoded correctly.

Reply:
Would a size header make sense?

Reply:
@jeffhostetler what do you think

Reply:
let's not do a size header/prefix.  i think we're good with the LF.